From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Sat, 23 Nov 2019 15:01:17 +0100
Subject: Strip test binaries by default

Bug: https://jira.mongodb.org/browse/SERVER-34079
Bug-Ubuntu: https://launchpad.net/bugs/1763119
Forwarded: no
Last-Update: 2018-03-23

This is needed to reduce build size to within Launchpad's limit.
---
 site_scons/site_tools/mongo_unittest.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/site_scons/site_tools/mongo_unittest.py b/site_scons/site_tools/mongo_unittest.py
index 2ad0f51..47a0308 100644
--- a/site_scons/site_tools/mongo_unittest.py
+++ b/site_scons/site_tools/mongo_unittest.py
@@ -23,7 +23,14 @@ def build_cpp_unit_test(env, target, source, **kwargs):
     libdeps = kwargs.get('LIBDEPS', [])
     libdeps.append( '$BUILD_DIR/mongo/unittest/unittest_main' )
 
+    linkflags = []
+    linkflags.extend(env.get('LINKFLAGS', []))
+    linkflags.extend(kwargs.get('LINKFLAGS', []))
+    linkflags.append('-Wl,--strip-debug')
+
     kwargs['LIBDEPS'] = libdeps
+    kwargs['LINKFLAGS'] = linkflags
+
     kwargs['INSTALL_ALIAS'] = ['tests']
 
     result = env.Program(target, source, **kwargs)
